File: src/main/java/com/dev/photoCatalog/model/Location.java
package com.dev.photoCatalog.model;

import jakarta.persistence.*;

@Entity
public class Location {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int locationID;

    @Column(nullable = false, length = 256)
    private String locationName;

    // getters and setters
    public int getLocationID() {
        return locationID;
    }

    public void setLocationID(int locationID) {
        this.locationID = locationID;
    }

    public String getLocationName() {
        return locationName;
    }

    public void setLocationName(String locationName) {
        this.locationName = locationName;
    }
}

File: frontend/src/models/Location.js
class Location {
    constructor(locationID, name, address) {
      this.locationID = locationID; 
      this.name = name; 
      this.address = address;
    }
  
    // Getters
    getLocationID() {
      return this.locationID;
    }
  
    getName() {
      return this.name;
    }
  
    getAddress() {
      return this.address;
    }
  
    // Setters
    setLocationID(locationID) {
      this.locationID = locationID;
    }
  
    setName(name) {
      this.name = name;
    }
  
    setAddress(address) {
      this.address = address;
    }
  }
  
  export default Location;
  